### DISCUSSION TITLE
Replay page

### DISCUSSION THREAD

**Discussion by sunehs** [elmadev/elmaonline-site#416] [Feature discussion]
For general discussion of the replay page: https://elma.online/r/2y37yp4n3n

This post will be updated to contain planned features and changes as well as ideas that needs to be fledged out. Discuss them below. If you want to work on one of these, create an issue if it doesn't exist and assign yourself.

**Planned**
- [x] Theater mode: Normal mode would look similar to current layout and theater mode would be be closer to [cup replay view](https://elma.online/r/cup/4370/Kopa10Stini)
- [x] Filename in url: Update url to something like https://elma.online/r/2y37yp4n3n/muiz8500weird (old urls should still work)
- [x] Replay tags: Filter other replays by tags instead of checkboxes
- [x] Show amount of ratings
- [x] Merge: Option to merge or not merge when clicking other replays
- [x] recplayer user settings: Grass, low quality etc.
- [x] recplayer inputs: Double click to fullscreen, pinch to zoom etc.

**Ideas**
- Share at: Generate a link with a timestamp that will start the replay from that time
- Show record/top times in level

**Comment by metsavir** [comment]
- other replays filtering by tags if "Nitro", "Bug" and "TAS" flags are replaced with tags
- amount of ratings next to stars

**Comment by j-masci** [comment]
- show world record time and player on non-level pages, such as battle, replay.
- icon/button to show svg image of level anywhere there is a recplayer.
- tag for non-featured replays, which show up under "other replays in level" but not on home page.

**Comment by PabloCorso** [comment]
For the ideas related to the recplayer features, I'm working on a quick refactor with possibly some rewrites for the RecPlayer react component, to have more control of the recplayer.

Can see the refactor at this fork: https://github.com/PabloCorso/recplayer-react

1. For Share at idea. Need access to the current recplayer frame to generate the timestamp. Later when the url is visited need to do the reverse, get the frame from the timestamp and initialize recplayer in that frame.
2. Merge replays. We could for example, allow a recUrl prop of RecPlayer component to be a single string or an array of string, one or multiple recs.
3. RecPlayer inputs. For example: allow access to the recplay controller at init, or at event handlers, etc. And then implement whatever we can with the current recplay controller.
